schema/query-log-schema-2g.json

```diff
@@ -0,0 +1,167 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "definitions": {
+    "unit": {
+      "type": "string",
+      "enum": [
+        "nanosecond",
+        "microsecond",
+        "millisecond",
+        "second",
+        "minute",
+        "hour",
+        "day",
+        "week",
+        "bit",
+        "decabit",
+        "hectobit",
+        "kilobit",
+        "megabit",
+        "gigabit",
+        "terabit",
+        "petabit",
+        "exabit",
+        "zettabit",
+        "yottabit",
+        "kibibit",
+        "mebibit",
+        "gibibit",
+        "tebibit",
+        "pebibit",
+        "exbibit",
+        "zebibit",
+        "yobibit",
+        "byte",
+        "decabyte",
+        "hectobyte",
+        "kilobyte",
+        "megabyte",
+        "gigabyte",
+        "terabyte",
+        "petabyte",
+        "exabyte",
+        "zettabyte",
+        "yottabyte",
+        "kibibyte",
+        "mebibyte",
+        "gibibyte",
+        "tebibyte",
+        "pebibyte",
+        "exbibyte",
+        "zebibyte",
+        "yobibyte",
+        "rotation",
+        "degree",
+        "radian",
+        "celsius",
+        "fahrenheit",
+        "kelvin"
+      ]
+    },
+    "sample": {
+      "type": "object",
+      "properties": {
+        "unitNumerators": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/unit"
+          }
+        },
+        "unitDenominators": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/unit"
+          }
+        },
+        "value": {
+          "type": "number"
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "additionalProperties": false
+    },
+    "metric": {
+      "type": "object",
+      "properties": {
+        "values": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/sample"
+          }
+        }
+      },
+      "required": [
+        "values"
+      ],
+      "additionalProperties": false
+    },
+    "metricsList": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/metric"
+      }
+    }
+  },
+  "title": "Query Log",
+  "description": "log file entry for ingestion by tsd aggregator",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
+    },
+    "date": {
+      "type": "string",
+      "format": "date-time"
+    },
+    "dimensions": {
+      "type": "object",
+      "properties": {          
+        "host": {
+          "type": "string"
+        },
+        "service": {
+          "type": "string"
+        },
+        "cluster": {
+          "type": "string"
+        }
+      },
+      "required": [
```

Comment:
That's a good question.  I think we're heading the way of not requiring any particular dimensions.  But cluster agg will throw a fit if there is no cluster dimension.  I'd say that perhaps the format shouldn't require them, but it's highly recommended that you add them since there may be other systems that require them.